pkg/controller/job/job_controller_test.go

```diff
@@ -3045,13 +3095,138 @@ func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
 					},
 				},
 			},
-			wantStatusActive: 1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusActive:      1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusTerminating: pointer.Int32(1),
+		},
+		"default value for podreplacementpolicy is Failed when not specifying PodReplacementPolicy": {
+			enableJobPodFailurePolicy:     true,
+			enablePodDisruptionConditions: true,
+			enableJobPodReplacementPolicy: true,
```

Comment:
Copy this test case when `enableJobPodReplacementPolicy=false`. We still want the behavior in that case. Also, make sure that the pods aren't recreated in this case. I guess it is tested by `wantStatusActive=0`, but we need to double check.